diff --git a/CMakeLists.txt b/CMakeLists.txt
index d9ad03c..1024e09 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -12,7 +12,7 @@ option(QIO_ENABLE_QMP_ROUTE  "Enable using QMP_route" OFF)
 option(QIO_ENABLE_OUTPUT_BUFFERING "Enable OutputBuffering" OFF)
 option(QIO_ENABLE_FAST_ROUTE "Enable James Osborns Fast DML route" OFF)
 option(QIO_ENABLE_SANITIZERS "Enable Undefined Behaviour and Address Sanitizers" OFF)
-option(QIO_BUILD_TESTS "Enable building of test programs" ON)
+option(QIO_BUILD_TESTS "Enable building of test programs" OFF)
 
 set(QIO_DML_BUF_BYTES "262144"  CACHE STRING "Maximum DML Buffer Size in bytes")
 set(QMP_DIR "" CACHE STRING "QMP Install Directory")
